### ISSUE TITLE
Keras feature DropConnect

### ISSUE THREAD

**Issue by autoih** [tensorflow/tensorflow#24009]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using):1.1.0 (Keras version: 2.2.2)
- Are you willing to contribute it (Yes/No):Yes

**Describe the feature and the current behavior/state.**
I would like to know if you would accept a pull request in ``DropConnect’’,i.e., https://cs.nyu.edu/~wanli/dropc/, which is a generalization of Hinton’s Dropout. 
In experiments, we may sometimes want to drop edges(weights) instead of neurons, I think it is very convenient to have the general version for regularizing layers. 
If so, I am willing to work on this. I would add a flag to the operator so that users can decide it to run. This changes should not be difficult to make.

**Will this change the current api? How?**
No. Probably based on current api. 

**Who will benefit with this feature?**
Users build their own architectures/algorithms.  
**Any Other info.**

**Comment by bhack** [comment]
https://github.com/keras-team/keras/pull/9898#issuecomment-380590161

**Comment by bionicles** [comment]
would use

**Comment by rmothukuru** [comment]
@autoih,
Sorry for the delayed response. As per this comment, [this Layer](https://github.com/keras-team/keras/pull/9898#issuecomment-380590161) might not be very useful. It has not attracted the attention of the Developers from the Community as well. 

If you feel this functionality is still relevant, please feel free to raise a PR. Thanks!
